### PR TITLE
Add CaretBrush Feature to Textbox

### PR DESCRIPTION
**Pull request by Abdulbaqi-Alshareef**
## What does the pull request do?
This PR adds CaretBrush property to TextBox to change the color of caret (blinking cursor when writing text in Textbox) like what we have in WPF

CaretBrush

Aslo adds example to show this feature  in Control Catalog

## What is the current behavior?
Now caret brush is depends on background color of TextBox to make some contrast, and if background has no color then it gets default color (black)

## What is the updated/expected behavior with this PR?
Caret color can be changed by setting color name or brush to CaretBrush styled property like this:
`<TextBox Text="Custom caret brush" CaretBrush="DarkOrange"/>`

## Breaking changes
no Breaking changes

## Fixed issues

Adds feature requested  here #2628

### CODE REVIEW

**Comment by Gillibald** [comment]
Can you add SelectionBrush as well? Currently this is customized by a named resource "HighlightBrush" that could be changed to use a styled property too. The default value could be "HighlightBrush"

File: src/Avalonia.Controls/Presenters/TextPresenter.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ public class TextPresenter : TextBlock
         public static readonly StyledProperty<char> PasswordCharProperty =
             AvaloniaProperty.Register<TextPresenter, char>(nameof(PasswordChar));
 
+        public static readonly StyledProperty<IBrush> CaretBrushProperty =
+         AvaloniaProperty.Register<TextPresenter, IBrush>(nameof(CaretBrushProperty));
```
Comment (Sorien) [inline src/Avalonia.Controls/Presenters/TextPresenter.cs]: there is wrong indent

File: src/Avalonia.Controls/TextBox.cs [hunk 2]
```diff
@@ -38,6 +38,9 @@ public class TextBox : TemplatedControl, UndoRedoHelper<TextBox.UndoRedoState>.I
         public static readonly StyledProperty<char> PasswordCharProperty =
             AvaloniaProperty.Register<TextBox, char>(nameof(PasswordChar));
 
+        public static readonly StyledProperty<IBrush> CaretBrushProperty =
+  AvaloniaProperty.Register<TextPresenter, IBrush>(nameof(CaretBrushProperty));
```
Comment (Sorien) [inline src/Avalonia.Controls/TextBox.cs]: same here

File: src/Avalonia.Controls/Presenters/TextPresenter.cs (hunk 1 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/Presenters/TextPresenter.cs]: It's ok now

File: src/Avalonia.Controls/TextBox.cs (hunk 2 above)
Comment (Abdulbaqi-Alshareef) [reply src/Avalonia.Controls/TextBox.cs]: It's ok now

**Comment by Abdulbaqi-Alshareef** [comment]
> 
> 
> Can you add SelectionBrush as well? Currently this is customized by a named resource "HighlightBrush" that could be changed to use a styled property too. The default value could be "HighlightBrush"
Its in another PR here #2654

**Pull request merged**